php/class-customize-snapshot-manager.php

```diff
@@ -123,6 +123,11 @@ public function __construct( Plugin $plugin ) {
 		add_action( 'customize_save_after', array( $this, 'save_snapshot' ) );
 		add_action( 'admin_bar_menu', array( $this, 'customize_menu' ), 41 );
 		add_action( 'customize_controls_print_footer_scripts', array( $this, 'render_templates' ) );
+		
+		add_filter( 'theme_mod_nav_menu_locations', array( $this, 'filter_theme_mod_nav_menu_locations' ) );
+		add_filter( 'wp_get_nav_menus', array( $this, 'filter_wp_get_nav_menus' ) );
+		add_filter( 'wp_get_nav_menu_items', array( $this, 'filter_wp_get_nav_menu_items' ), 10, 3 );
+		add_filter( 'wp_get_nav_menu_object', array( $this, 'filter_wp_get_nav_menu_object' ), 10, 2 );
```

Comment:
Update: the issue is that these filters are used only settings-specifically but the menus are initially loaded at this moment in `WP_Customize_Nav_Menus`:
https://github.com/xwp/wordpress-develop/blob/master/src/wp-includes/class-wp-customize-nav-menus.php#L509

Loading the `Customize_Nav_Menu_Setting` by itself to use the filters would require being related to existing setting IDs which are unknown before loading the Snapshot data. Any ideas how to still use these instead of creating new filters? Otherwise it seems that new filters are needed.

There's a second issue as well. The following function doesn't return a correct value when used with negative menu ID-s:
https://github.com/xwp/wordpress-develop/blob/57d2c46591e6593c8d6bfcb2100b6538e80f9dce/src/wp-includes/customize/class-wp-customize-nav-menu-section.php#L37-L42
E.g. `nav_menu[-123]` returns 0.